### ISSUE TITLE
nothing renders

### ISSUE THREAD

**Issue by mouadbouras** [mbasso/asm-dom-boilerplate#4]
I followed the instructions to build and run the application. Everything seems to build properly but when I run the command "npm start" localhost:1234/ loads with a blank html page. 
When I inspect the page it only has one empty div element with the "root" ID. 
I am new to wasm so I'm not sure if this is the expected behavior. 

If it is, then how do I display dom elements from the C++ file? 

Thanks.

**Comment by mbasso** [comment]
Hi @mouadbroos,
the boilerplate should automatically render a counter in the center of the page, so it seems that there is a problem... Could you please run these commands and paste here the versions?

```bash
node -v
npm -v
make -v
emcc -v
```

Could you please also provide the versions of your OS and browser?
Are there some errors in the browser console? It also might be useful the output of the terminal

**Comment by mouadbouras** [comment]
Command outputs:
```
>node -v
v8.11.2

>npm -v
5.6.0

>make -v
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-pc-mingw32

>emcc -v
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 1.38.3
clang version 6.0.1  (emscripten 1.38.3 : 1.38.3)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Users\---\Asmdom\emsdk\clang\e1.38.3_64bit
INFO:root:(Emscripten: Running sanity checks)
```
OS Version: 
I'm using Windows 10 x64 and latest Chrome/Edge
There are no console errors!! 


Console Output : 

```
>npm install
npm WARN asm-dom-boilerplate@0.1.0 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

up to date in 5.329s

>npm start

> asm-dom-boilerplate@0.1.0 start C:\Users\---\Asmdom\asm-dom-boilerplate
> make start
ERROR: Invalid argument/option - 'P:/'.
Type "FORFILES /?" for usage.
npx concurrently \
                -p "[{name}]" \
                -n "Parcel,Make" \
                -c "bgGreen.bold,bgBlue.bold" \
                "npx cross-env NODE_ENV=development parcel index.html --open" \
                "nodemon --exec \"make dist/wasm dist/asmjs\""
npx: installed 1 in 1.965s
Path must be a string. Received undefined
C:\User\---\Asmdom\asm-dom-boilerplate\node_modules\concurrently\src\main.js
[Make] [nodemon] 1.17.4
[Make] [nodemon] reading config .\nodemon.json
[Make] [nodemon] to restart at any time, enter `rs`
[Make] [nodemon] or send SIGHUP to 158612 to restart
[Make] [nodemon] ignoring: .\.git/**/* .\glue/**/* .\dist/**/* .\.cache/**/* .\temp/**/* node_modules/**/node_modules
[Make] [nodemon] watching: C:\Users\---\Asmdom\asm-dom-boilerplate\src/**/*
[Make] [nodemon] watching extensions: c,cpp,h,hpp
[Make] [nodemon] starting `make dist/wasm dist/asmjs`
[Make] [nodemon] spawning
[Make] [nodemon] child pid: 157864
[Make] [nodemon] watching 3 files
[Make] ERROR: Invalid argument/option - 'P:/'.
[Make] Type "FORFILES /?" for usage.
[Make] make[1]: Entering directory `C:/Users/---/Asmdom/asm-dom-boilerplate'
[Make] make[1]: `dist/wasm' is up to date.
[Make] make[1]: `dist/asmjs' is up to date.
[Make] make[1]: Leaving directory `C:/Users/---/Asmdom/asm-dom-boilerplate'
[Make] [nodemon] clean exit - waiting for changes before restart
[Parcel] npx: installed 1 in 1.487s
[Parcel] Path must be a string. Received undefined
[Parcel] C:\Users\---\Asmdom\asm-dom-boilerplate\node_modules\cross-env\dist\bin\cross-env.js
[Parcel] Server running at http://localhost:1234
√  Built in 12.61s.
[Parcel]
```

**Comment by mbasso** [comment]
It seems that there is a problem with the `FORFILES` command at line 3 of the `Makefile`. I've tried but I'm unable to reproduce it. 

```
ERROR: Invalid argument/option - 'P:/'.
Type "FORFILES /?" for usage.
```

Can you please copy here the first 6 lines of your `Makefile`?

**Comment by mouadbouras** [comment]
Here you go! 
all I did was uncomment lines 4 and 5 and comment out line 3.
```
# if you are using windows, please comment line 3 and uncomment line 4 and 5
SRCDIR := src
#SRCS := $(shell find $(SRCDIR) -name "*.cpp")
SRCS := $(patsubst "%",%,$(shell FORFILES /P "$(SRCDIR)" /S /M *.cpp /C "CMD /C ECHO @relpath"))
SRCS := $(subst \,/,$(SRCS:.\\%=$(SRCDIR)\\%))
```

**Comment by mbasso** [comment]
That's really strange, line 4 should run something like this:
`FORFILES /P "src" /S /M *.cpp /C "CMD /C ECHO @relpath"`

I don't get why it is complaining about a `P:/`. If you try to run the command above in the terminal all works fine?

**Comment by mouadbouras** [comment]
I'm not sure either! :-( 

When I run the command in terminal it gives me : 

```
>FORFILES /P "src" /S /M *.cpp /C "CMD /C ECHO @relpath"

".\index.cpp"
```

**Comment by mbasso** [comment]
mmm this seems to work fine... I have to understand the cause of this issue.
Tomorrow I'll go on holiday for 3 days, I'll check it again on monday night.

I know that it's not the best and it is only a workaround, but, until that day,if you want to try this boilerplate, you can change your `Makefile` in this way:

```
# if you are using windows, please comment line 3 and uncomment line 4 and 5
SRCDIR := src
#SRCS := $(shell find $(SRCDIR) -name "*.cpp")
#SRCS := $(patsubst "%",%,$(shell FORFILES /P "$(SRCDIR)" /S /M *.cpp /C "CMD /C ECHO @relpath"))
#SRCS := $(subst \,/,$(SRCS:.\\%=$(SRCDIR)\\%))
SRCS := src/index.cpp
```

After that `npm start` should work fine. Basically you have to add you `.cpp` files inside the `src` folder manually in the `Makefile`, for example:

```
SRCS := \
	src/index.cpp \
        src/file1.cpp \
        src/file2.cpp
```

Please note that you have to relaunch `npm start` if you add or delete some of them. If you just change them the boilerplate should automatically recompile your app to wasm.

I've tried this repo on my windows 10 and all works fine so it's hard to identify the cause... I'm sorry for this issue, I hope to find a fix soon!

**Comment by mbasso** [comment]
Hi @mouadbroos,
news about this?

**Comment by F1r3Hydr4nt** [comment]
I also have nothing rendering and receive a console output in Chrome:
![screen shot 2018-10-19 at 17 24 20](https://user-images.githubusercontent.com/3375175/47231180-20ac1780-d3c4-11e8-829b-293bccd3471f.png)

In Firefox it's much less verbose: TypeError: second argument must be an object

Furthermore I needed to circumvent another error:
asm-dom-boilerplate/glue/load.js:4:11: Cannot resolve dependency '../dist/wasm/app.js' at '/Users/XX/simple_pull/asm-dom-boilerplate/dist/app.js'

By changing '.js' to '.wasm' and I think it was correct? <---**Scratch this!**

*In bold above was my problem! Nice entry point to wasm, thanks.

**Comment by mbasso** [comment]
Hi @freddiehonohan,
can you please run the following commands and paste here the output?

```bash
node -v
npm -v
make -v
emcc -v
java -version
```

Have you run `npm install` and then `npm start`?

**Comment by F1r3Hydr4nt** [comment]
Hey dude, no I got it to work thanks.

At some stage the build crashed and left my app.js in limbo with an app.js.mem so then I tried working around it but that actually caused more errors. What fixed it was a 'make' then an npm start which rebuilt a fresh app.js.

I'm fiddling about with a service worker now, trying to get it to pass the Google Lighthouse audit. Any ideas how to add a manifest file to the make list?
